core/trino-main/src/test/java/io/trino/block/BlockAssertions.java

```diff
@@ -596,4 +632,223 @@ public static RunLengthEncodedBlock createRLEBlock(long value, int positionCount
+        else if (type == VARCHAR) {
+            block = createRandomStringBlock(positionCount, primitiveNullRate, MAX_STRING_SIZE);
+        }
+        else {
+            block = createRandomBlockForNestedType(type, positionCount, primitiveNullRate, nestedNullRate, wrappings);
+        }
+
+        if (!wrappings.isEmpty()) {
+            block = wrapBlock(block, positionCount, wrappings);
+        }
+
+        return block;
+    }
+
+    private static Block createRandomBlockForNestedType(Type type, int positionCount, float primitiveNullRate, float nestedNullRate, List<Encoding> wrappings)
+    {
+        Block block;
+        // Builds isNull and offsets of size positionCount
+        boolean[] isNull = null;
+        Set<Integer> nullPositions = null;
+        if (nestedNullRate > 0) {
+            isNull = new boolean[positionCount];
+            nullPositions = chooseNullPositions(positionCount, nestedNullRate);
+        }
+        int[] offsets = new int[positionCount + 1];
+
+        for (int position = 0; position < positionCount; position++) {
+            if (nestedNullRate > 0 && nullPositions.contains(position)) {
+                isNull[position] = true;
+                offsets[position + 1] = offsets[position];
+            }
+            else {
+                // RowType doesn't need offsets, so we just use 1,
+                // for ArrayType and MapType we choose randomly either array length or map size at the current position
+                offsets[position + 1] = offsets[position] + (type instanceof RowType ? 1 : RANDOM.nextInt(ENTRY_SIZE) + 1);
+            }
+        }
+
+        // Builds the nested block of size offsets[positionCount].
+        if (type instanceof ArrayType) {
+            Block valuesBlock = createRandomBlockForType(((ArrayType) type).getElementType(), offsets[positionCount], primitiveNullRate, nestedNullRate, wrappings);
+            block = fromElementBlock(positionCount, Optional.ofNullable(isNull), offsets, valuesBlock);
+        }
+        else if (type instanceof MapType) {
+            MapType mapType = (MapType) type;
+            Block keyBlock = createRandomBlockForType(mapType.getKeyType(), offsets[positionCount], 0.0f, 0.0f, wrappings);
+            Block valueBlock = createRandomBlockForType(mapType.getValueType(), offsets[positionCount], primitiveNullRate, nestedNullRate, wrappings);
+
+            block = mapType.createBlockFromKeyValue(Optional.ofNullable(isNull), offsets, keyBlock, valueBlock);
+        }
+        else if (type instanceof RowType) {
+            List<Type> fieldTypes = type.getTypeParameters();
+            Block[] fieldBlocks = new Block[fieldTypes.size()];
+
+            for (int i = 0; i < fieldBlocks.length; i++) {
+                fieldBlocks[i] = createRandomBlockForType(fieldTypes.get(i), positionCount, primitiveNullRate, nestedNullRate, wrappings);
+            }
+
+            block = fromFieldBlocks(positionCount, Optional.ofNullable(isNull), fieldBlocks);
```

Comment:
do not static import this method, static name is not sufficient to recognize it's from `RowBlock`